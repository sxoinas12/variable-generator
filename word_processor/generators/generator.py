import spacy
from py.error import Error

nlp = spacy.load("en_core_web_trf")


class Generator():
    """
        Generator class that will use a strategy resolver to generate suggestions for a text
        :return:

    """

    # passed text
    text = None
    # parsed text from spacy lib
    doc = None
    # the strategy resolver to use
    strategy_resolver = None
    # the parser for the presentation
    parser = None

    # suggestions calculated
    suggestions = None

    def __init__(self, text=None, strategy=None, parser=None):
        self.strategy_resolver = strategy
        self.parser = parser
        if text is not None:
            self.set_text(text)

    def suggest(self):
        if self.doc is None or self.text is None:
            raise Error('No document')

        self.suggestions = self.strategy_resolver(self.doc)

        return self.suggestions

    def set_text(self, text):
        self.text = text
        self.doc = nlp(text)

    def set_strategy(self, strategy_resolver):
        self.strategy_resolver = strategy_resolver

    def present(self):
        self.parser.suggestions = self.suggestions
        return self.parser.present()