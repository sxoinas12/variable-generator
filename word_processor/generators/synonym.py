import nltk
from nltk.corpus import wordnet
import os

from nltk.corpus.reader import WordNetError

print("Loading nltk", os.path.dirname(__file__) + '/../../nltk_data/')
nltk.data.path.append(os.path.dirname(__file__) + '/../../nltk_data/')


def get_synonyms(word):
    synonyms = []

    try:
        synsets_words = wordnet.synsets(word)
    except WordNetError:
        return []

    for syn in synsets_words:
        for lemma in syn.lemmas():
            synonyms.append(lemma.name())

    return synonyms


def get_antonyms(word):
    antonyms = []

    try:
        synsets_words = wordnet.synsets(word)
    except WordNetError:
        return []

    for syn in synsets_words:
        for lemma in syn.lemmas():
            if lemma.antonyms():
                for antonym in lemma.antonyms():
                    antonyms.append(antonym.name())

    return antonyms