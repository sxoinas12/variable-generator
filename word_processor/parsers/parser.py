from spacy.tokens import Token


class Parser:
    suggestions = None
    parsed_suggestions = None

    def __init__(self, suggestions=None):
        self.suggestions = suggestions

    def parse_word(self, word: Token):
        return word.text

    def parse_words(self, words: [Token]):
        return [self.parse_word(word) for word in words]

    def parse(self):
        self.parsed_suggestions = [self.parse_words(words) for words in self.suggestions]

        return self

    def present_word(self, word: str):
        return word

    def present_words(self, words: [str]):
        presented_words = [self.present_word(word) for word in words]

        return ''.join(presented_words)

    def present(self):
        self.parse()

        return [self.present_words(words) for words in self.parsed_suggestions]
