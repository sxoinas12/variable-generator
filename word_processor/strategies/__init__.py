from .leaf_strategy import leaf_strategy
from .leaf_no_preps_strategy import leaf_no_prep_strategy
from .adverbial_strategy import adverbial_strategy
from .compound_strategy import compound_strategy
from .flip_prep_pobj_strategy import flip_prep_pobj_strategy
from .back_and_forth_strategy import back_and_forth_strategy

__all__ = [
    'leaf_strategy',
    'leaf_no_prep_strategy',
    'adverbial_strategy',
    'compound_strategy',
    'flip_prep_pobj_strategy',
    'back_and_forth_strategy',
]
