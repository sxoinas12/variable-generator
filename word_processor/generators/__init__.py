from .generator import Generator
from .synonym import get_synonyms, get_antonyms

__all__ = [
    'Generator',
    'get_antonyms',
    'get_synonyms',
]
