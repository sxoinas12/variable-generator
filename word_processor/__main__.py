import sys

from word_processor.generators import Generator
from word_processor.parsers import JavascriptParser
from word_processor.strategies import back_and_forth_strategy


def calculate_names(text: str):
    generator = Generator(text=text, strategy=back_and_forth_strategy, parser=JavascriptParser())

    generator.suggest()

    presentation = generator.present()

    print('\n\n'.join(presentation))


# Press the green button in the gutter to run the script.
if __name__ == '__main__':
    calculate_names(sys.argv[1])

# See PyCharm help at https://www.jetbrains.com/help/pycharm/
