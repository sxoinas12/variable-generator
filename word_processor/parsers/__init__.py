from .parser import Parser
from .javascript import JavascriptParser

__all__ = [
    'Parser',
    'JavascriptParser',
]