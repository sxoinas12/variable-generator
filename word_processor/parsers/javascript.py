from spacy.tokens import Token

from word_processor.parsers import Parser


class JavascriptParser(Parser):

    def __init__(self, **kwargs):
        super().__init__(**kwargs)

    def parse_word(self, word: Token):
        return word.lemma_.title()

    def parse_words(self, words: [Token]):
        parsed_words = super().parse_words(words)

        return parsed_words
