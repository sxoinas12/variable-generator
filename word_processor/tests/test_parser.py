import json

import pytest

from word_processor.generators import Generator
from word_processor.parsers import Parser
from word_processor.strategies import leaf_strategy

@pytest.mark.parametrize('text', [
    'Regex for redacted phone numbers with extra info for PlayStation'
])
def test_parser(snapshot, text):
    generator = Generator(text=text, strategy=leaf_strategy, parser=Parser())

    generator.suggest()

    presentation = generator.present()

    assert presentation == snapshot

