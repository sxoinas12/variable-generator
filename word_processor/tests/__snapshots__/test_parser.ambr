# name: test_parser[Regex for redacted phone numbers with extra info for PlayStation]
  <class 'list'> [
    'Regexfornumbersredacted',
    'Regexfornumbersphone',
    'Regexfornumberswithinfoextra',
    'RegexfornumberswithinfoforPlayStation',
  ]
---
